### PR TITLE
PE-53 | Created a View Model for Profile Page

### PR DESCRIPTION
**Pull request by tonynguyen98**
**What I Did:** 
- Added a `findByUserId` to both `QuestionRepository` and `AnswerRepository`.
- Created a new Profile Object that contains data from `User`, `Question` and `Answer`.
- Create the service that will aggregate `User`, `Questions` (by userId ) and `Answers` (by userId) into a single model for the front-end.
- Created a controller that will send this model out via HTTP GET.

**How to Test:**
1. Create a new User using the HTTP POST controller created by @ayeejane.
```json
{
"firstName" : "Jennie",
"lastName" : "Kim",
"firebaseUuid" : "123456",
"email" : "jenniekim@yahoo.com",
"city" : "Los Angeles",
"category" : ["Technology"],
"pictureUrl" : "hellokitty.com",
"role" : 0,
"rating" : 9
}
```
2. Create a couple Questions using HTTP POST controller created by @dominikyz.  IMPORTANT: Make sure you use the id provided in step 1 for the `user_id` in Questions.
```json
{
    "title": "TestRecord2",
    "questionDate": "2012-04-23T18:25:43.511Z",
    "description": "Description goes here",
    "category": "science",
    "status": 1,
    "visible": 1,
    "userId": "ID FROM STEP 1"
}
```
3. Now, create a couple more Questions, but use a random `user_id`!!!
4. Create a couple Answers using HTTP POST controller created by @davidsquines.  IMPORTANT: Make sure you use the id provided in step 1 for the `user_id` in Answers.
```json
{
    "mediaUrls": [
        "www.google.com", 
        "www.csun.edu"
        ],
    "answerDate": "2012-04-23T18:25:43.511+00:00",
    "questionId": 7,
    "userId": "ID FROM STEP 1"
}
```
5. Now, create a couple more Answers, but use a random `user_id`!!!
6. Using Postman, hit HTTP GET `localhost:8080/v1/profile/{USER ID}`. You should a `ok` response that will return the user and questions/answers that contain only the correct `user_id`!.

### CODE REVIEW

**Comment by txt020** [comment]
@tonynguyen98 For the PUT request, is the URL the same as the GET request?

**Comment by tonynguyen98** [comment]
> @tonynguyen98 For the PUT request, is the URL the same as the GET request?

Apologies, I meant POST request and the routes for those would be `localhost:8080/v1/users`, `localhost:8080/v1/questions` and `localhost:8080/v1/answers`. In that order respectively.

**Review by dominikyz — APPROVED** [review]
Code looks great. CRUD operations work like a charm! Awesome job!

File: src/main/java/com/ProjectEureka/backend/services/ProfileServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    AnswerRepository answerRepository;
+
+    @Override
+    public Profile getProfile(String id) {
+        Optional<User> user = userRepository.findById(id);
+        List<Question> questionsByUserId = questionRepository.findByUserId(id);
+        List<Answer> answersByUserId = answerRepository.findByUserId(id);
+
+        if (user.isPresent() && !questionsByUserId.isEmpty() && !answersByUserId.isEmpty()) {
+            return new Profile(user.get(), questionsByUserId, answersByUserId);
```
Comment (ayeejane) [inline src/main/java/com/ProjectEureka/backend/services/ProfileServiceImpl.java]: A profile might not have both questions and answers

**Review by ayeejane — CHANGES_REQUESTED** [review]

**Review by davidsquines — APPROVED** [review]
Works as intended

**Review by txt020 — APPROVED** [review]
Everything looks good and it works as intended.

**Review by ayeejane — APPROVED** [review]
Works beautifully now!

**Pull request merged**